src/rust/engine/fs/src/lib.rs

```diff
@@ -297,64 +381,131 @@ impl PathGlob {
   }
 }
 
+#[derive(Debug)]
+pub enum StrictGlobMatching {
+  Error,
+  Warn,
+  Ignore,
+}
+
+impl StrictGlobMatching {
+  // TODO(cosmicexplorer): match this up with the allowed values for the GlobMatchErrorBehavior type
+  // in python somehow?
+  pub fn create(behavior: &str) -> Result<Self, String> {
+    match behavior {
+      "ignore" => Ok(StrictGlobMatching::Ignore),
+      "warn" => Ok(StrictGlobMatching::Warn),
+      "error" => Ok(StrictGlobMatching::Error),
+      _ => Err(format!(
+        "Unrecognized strict glob matching behavior: {}.",
+        behavior,
+      )),
+    }
+  }
+
+  pub fn should_check_glob_matches(&self) -> bool {
+    match self {
```

Comment:
If you derive `Eq`/`PartialEq` for this enum, you can do `&StrictGlobMatching::Ignore == self`.